src/Runner.Sdk/Util/UrlUtil.cs

```diff
@@ -6,9 +6,16 @@ public static class UrlUtil
     {
         public static bool IsHostedServer(UriBuilder gitHubUrl)
         {
-            return string.Equals(gitHubUrl.Host, "github.com", StringComparison.OrdinalIgnoreCase) ||
+            if (StringUtil.ConvertToBoolean(Environment.GetEnvironmentVariable("GITHUB_ACTIONS_RUNNER_FORCE_GHES")))
+            {
+                return false;
+            }
+
+            return 
+                string.Equals(gitHubUrl.Host, "github.com", StringComparison.OrdinalIgnoreCase) ||
                 string.Equals(gitHubUrl.Host, "www.github.com", StringComparison.OrdinalIgnoreCase) ||
-                string.Equals(gitHubUrl.Host, "github.localhost", StringComparison.OrdinalIgnoreCase);
+                string.Equals(gitHubUrl.Host, "github.localhost", StringComparison.OrdinalIgnoreCase) ||
+                gitHubUrl.Host.EndsWith(".ghe.com", StringComparison.OrdinalIgnoreCase);
```

Comment:
Should there be an equivalent `ghe.localhost` check since we use that in github/github development?

Reply:
That is a good point. I hadn't thought about local development. We can track keep tract of that and add it in. Thanks @mtodd!

cc @javs-perez